supervisor/task.go

```diff
@@ -1,89 +1,29 @@
 package supervisor
 
 import (
-	"os"
-	"time"
+	"sync"
 
 	"github.com/docker/containerd/runtime"
-	"github.com/opencontainers/specs"
 )
 
-type TaskType string
-
-const (
-	ExecExitTaskType         TaskType = "execExit"
-	ExitTaskType             TaskType = "exit"
-	StartContainerTaskType   TaskType = "startContainer"
-	DeleteTaskType           TaskType = "deleteContainerEvent"
-	GetContainerTaskType     TaskType = "getContainer"
-	SignalTaskType           TaskType = "signal"
-	AddProcessTaskType       TaskType = "addProcess"
-	UpdateContainerTaskType  TaskType = "updateContainer"
-	UpdateProcessTaskType    TaskType = "updateProcess"
-	CreateCheckpointTaskType TaskType = "createCheckpoint"
-	DeleteCheckpointTaskType TaskType = "deleteCheckpoint"
-	StatsTaskType            TaskType = "events"
-	OOMTaskType              TaskType = "oom"
-)
-
-func NewTask(t TaskType) *Task {
-	return &Task{
-		Type:      t,
-		Timestamp: time.Now(),
-		Err:       make(chan error, 1),
-	}
-}
-
 type StartResponse struct {
 	Container runtime.Container
 }
 
-type Task struct {
-	Type          TaskType
-	Timestamp     time.Time
-	ID            string
-	BundlePath    string
-	Stdout        string
-	Stderr        string
-	Stdin         string
-	Console       string
-	Pid           string
-	Status        int
-	Signal        os.Signal
-	Process       runtime.Process
-	State         runtime.State
-	ProcessSpec   *specs.Process
-	Containers    []runtime.Container
-	Checkpoint    *runtime.Checkpoint
-	Err           chan error
-	StartResponse chan StartResponse
-	Stat          chan *runtime.Stat
-	CloseStdin    bool
-	ResizeTty     bool
-	Width         int
-	Height        int
-	Labels        []string
-}
-
-type Handler interface {
-	Handle(*Task) error
+type Task interface {
+	Error() chan error
```

Comment:
type Task interface {
 	ErrorCh() chan error
}

An odd definition for Task. Task is a name that hints that the implementers will run an arbitrary task (implement a method such as Action), can we rename this interface?

Also, in Go, all interfaces should be named after their methods, such as
```
type Handler interface {
    Handle()
}
```